libs/books/data-access/src/lib/+state/reading-list.effects.spec.ts

```diff
@@ -42,4 +60,92 @@ describe('ToReadEffects', () => {
       httpMock.expectOne('/api/reading-list').flush([]);
     });
   });
+  
+  describe('addBook$', () => {
+    it('should add a book to the reading list successfully', fakeAsync(() => {
+      actions.next(ReadingListActions.addToReadingList({ book, add: false  }));
+
+      effects.addBook$.subscribe(action => {
+        expect(action).toEqual(
+          ReadingListActions.confirmedAddToReadingList({ book, add: false })
+        );
+      })
+
+      httpMock.expectOne('/api/reading-list').flush({});
+    }));
+
+
+    it('should undo the added book when API returns error', fakeAsync(() => {
+      actions.next(ReadingListActions.addToReadingList({ book, add: false}));
+
+      effects.addBook$.subscribe(action => {
```

Comment:
This observable should be unsubscribed from when finished

Reply:
Resolved